src/main/java/Tasks/Deadline.java

```diff
@@ -0,0 +1,42 @@
+package Tasks;
+
+import Exceptions.InvalidFormatException;
+import Exceptions.InvalidInputException;
+import Exceptions.LeluException;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Deadline extends Task {
+    protected LocalDateTime deadline;
+    private DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+    public Deadline(String taskName, String deadline) throws DateTimeParseException {
+        super(taskName);
+        this.deadline = LocalDateTime.parse(deadline, formatter);
+
+    }
+
+    public static Deadline DeadlineOf(String input) throws InvalidInputException {
```

Comment:
Any reason why the checking of deadline details is done here? Not sure I like that the checking of details is done here instead of the AddDeadlineCommand or Parser. On top of that, the constructor of Deadline is also public. Same for the Event and ToDo classes.